components/nif/nifstream.hpp

```diff
@@ -84,12 +84,23 @@ class NIFStream {
     ///This is special since the version string doesn't start with a number, and ends with "\n"
     std::string getVersionString();
 
-    void getShorts(std::vector<short> &vec, size_t size);
-    void getFloats(std::vector<float> &vec, size_t size);
-    void getVector2s(std::vector<Ogre::Vector2> &vec, size_t size);
-    void getVector3s(std::vector<Ogre::Vector3> &vec, size_t size);
-    void getVector4s(std::vector<Ogre::Vector4> &vec, size_t size);
-    void getQuaternions(std::vector<Ogre::Quaternion> &quat, size_t size);
+    //Templated functions to handle reads
+    template <typename T>
+    T get(){throw std::runtime_error("Can not get this type of data from a NIF File!");}
```

Comment:
For facilitation of debugging it is better instead of `"this type"` to place in the exception message information on specific type, through `typeid(T).name()`.